pkg/api/meta/restmapper.go

```diff
@@ -468,91 +468,59 @@ func (o resourceByPreferredGroupVersion) Less(i, j int) bool {
 // RESTClient should use to operate on the provided group/kind in order of versions. If a version search
 // order is not provided, the search order provided to DefaultRESTMapper will be used to resolve which
 // version should be used to access the named group/kind.
-// TODO: consider refactoring to use RESTMappings in a way that preserves version ordering and preference
 func (m *DefaultRESTMapper) RESTMapping(gk schema.GroupKind, versions ...string) (*RESTMapping, error) {
+	mappings, err := m.commonRESTMappings(gk, versions...)
+	if err != nil {
+		return nil, err
+	}
+	if len(mappings) == 0 {
+		return nil, &NoKindMatchError{PartialKind: gk.WithVersion("")}
+	}
+	return mappings[0], nil
+}
+
+// RESTMappings returns the RESTMappings for the provided group kind in a rough internal preferred order. If no
+// kind is found it will return a NoResourceMatchError.
+func (m *DefaultRESTMapper) RESTMappings(gk schema.GroupKind) ([]*RESTMapping, error) {
```

Comment:
You mean change the interface ? https://github.com/kubernetes/kubernetes/blob/master/pkg/api/meta/interfaces.go#L181